src/components/Label/Label.tsx

```diff
@@ -110,8 +114,19 @@ class Label extends UIComponent<any, any> {
         },
       )
 
+      const imageElement = Image.create(
+        {
+          className: classes.image,
+          ...(typeof image === 'string' ? { src: image } : { ...image }),
+        },
```

Comment:
This does not handle other cases of shorthand, such as passing elements.  Instead of attempting to figure out which type of shorthand was passed, you can leave this to the factory.  To pass additional defaultProps, such as the className in this case, you can use the options object:


```jsx
Image.create(image, {
  generateKey: false,
  defaultProps: { className: classes.image },
})
```

Reply:
Also, it may be required to override the user's props and merge the `classes.image` with their `className`.  This way, if they pass a `props.className`, it doesn't override the styling classes:

```jsx
handleImageOverrideProps = props => ({
  ...props,
  className: cx(classes.image, props.className),
})

Image.create(image, {
  generateKey: false,
  overrideProps: this.handleImageOverrideProps,
})
```

Now, the component will override any props.className provided by the user by merging the style className into them.

***

Bonus: What if the user doesn't want our classNames?!

Answer: They can use a function for shorthand and whatever they want:

```jsx
<Label
  image={(Image, props) => <Image {...props} className='only these' />}
/>